apollo-api/src/main/java/com/spotify/apollo/RequestMetadata.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * -\-\-
+ * Spotify Apollo API Interfaces
+ * --
+ * Copyright (C) 2013 - 2016 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.apollo;
+
+import java.util.Optional;
+
+/**
+ * Describes an API for retrieving metadata about an incoming request.
+ */
+public interface RequestMetadata {
+
+  /**
+   * Returns the class that is the source of the metadata; this enables users to more easily
+   * understand where the metadata comes from and how to interpret data like the {@link #protocol()}.
+   */
+  Class<?> source();
```

Comment:
Seems implementation dependant if all metadata properties are added in the same place or not. Even more so if we add more properties to RequestMetadata. If client really need to depend on the creator of the metadata, maybe RequestMetadata could be implemented by a class that the client can downcast to?